### PR TITLE
[License] Simplify the Texture license to be pure Apache 2 (removing ASDK-Licenses).

### PR DESCRIPTION
**Pull request by appleguy**
With permission of the Facebook Open Source team, we are simplifying the Texture license, so that clients can rely on the terms of the Apache 2 license for all code in the Texture project.

This means that code originally forked from AsyncDisplayKit will be re-licensed from "BSD 3-clause + PATENTS v2" to Apache 2 (which includes a built-in patents grant).

Summarizing the change:

- The ASDK-Licenses directory is deleted, which removes the AsyncDisplayKit-only BSD license and expands the scope of Texture's original Apache 2 license.
- The header comments are simplified, but maintain the Facebook copyright on the original AsyncDisplayKit source files
- See the header comments [before and after](https://github.com/TextureGroup/Texture/pull/1077/files#diff-16417461de376720abfe05ef4cb7d73a)

There may be a bit of iteration to get everything right since this is such a large change. @JoelMarcey is the Facebook representative who is authorized to review this change.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ Before submitting a pull request, please make sure the following is done…
 6. If you haven't already, sign the CLA.
 
 **Copyright Notice for files**
-Copy and paste this to the top of your new file(s):
+Ensure this license header is included on all files in the repository, including new ones:
```
Comment (JoelMarcey) [inline CONTRIBUTING.md]: So no license mentions in the headers of code files?

File: README.md [hunk 2]
```diff
@@ -44,4 +44,4 @@ We welcome any contributions. See the [CONTRIBUTING](https://github.com/textureg
 
 ## License
 
-The Texture project was created by Pinterest as a continuation, under a different name and license, of the AsyncDisplayKit codebase originally developed by Facebook.  AsyncDisplayKit was originally released by Facebook under a BSD license and additional patent grant.  Those BSD and patent licenses govern use of code in Texture contributed prior to 4/13/2017 (the original AsyncDisplayKit code), and copies of the licenses are included in the root directory of this source tree for reference.  All code contributed to Texture after 4/13/2017 is released by Pinterest under an Apache 2.0 license.
+The Texture project is available for free use, as described by the Apache 2.0 license. Texture is based on code from the AsyncDisplayKit project originally developed by Facebook. Although AsyncDisplayKit was originally released by Facebook under a BSD license and additional "PATENTS v2" grant, the entire Texture codebase is now under a unified Apache 2 license with approval from Facebook to remove the PATENTS v2 file (including for the AsyncDisplayKit fork from 4/13/17).
```
Comment (JoelMarcey) [inline README.md]: Not sure we need the second sentence any longer. I think you can remove all references to the old license.

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR modifies CI file</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by Adlai-Holler** [comment]
Good branch name 😁

**Comment by JoelMarcey** [comment]
I am just clarifying a couple more things, but we are close to being able to merge this.

**Comment by nguyenhuy** [comment]
One small detail regarding this line: "you may not use this file except in compliance with the License." It's removed under the new license. I thought it's standard/should be included for Apache 2.0?

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big PR, please consider splitting it up to ease code review.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/6ac66e5fb34271c2b4ff5edc2a4a3e37/raw/9700ee57d06026d3c1954807658286326e5e812e/log.txt)

**Comment by maicki** [comment]
@JoelMarcey Thanks for letting us know - will merge now.

**Pull request merged**